### PR TITLE
[WFCORE-3071] add test coverage for SCRAM-*-PLUS Elytron SASL mechanisms

### PR DESCRIPTION
**Pull request by kwart**
JIRAs:
* upstream: https://issues.jboss.org/browse/WFCORE-3071
* downstream: https://issues.jboss.org/browse/JBEAP-12176

Add test coverage for Scram PLUS mechanisms. The tests will also serve as regression tests for https://issues.jboss.org/browse/JBEAP-11396.

### CODE REVIEW

File: testsuite/elytron/src/test/java/org/wildfly/test/integration/elytron/sasl/mgmt/ScramPlusMgmtSaslTestCase.java [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+                .build();
+        AuthenticationContext.empty().with(MatchRule.ALL, authCfg).withSsl(MatchRule.ALL, ssl)
+        .run(() -> assertAuthenticationFails("Connection through unexisting SASL mechanism should fail."));
+    }
+
+    /**
+     * Tests that authentication is possible using SCRAM-SHA-1.
+     */
+    @Test
+    @Ignore("JBEAP-11396")
```
Comment (dmlloyd) [inline testsuite/elytron/src/test/java/org/wildfly/test/integration/elytron/sasl/mgmt/ScramPlusMgmtSaslTestCase.java]: Maybe we should wait on this until we resolve the related discussion?

Comment (bstansberry) [inline testsuite/elytron/src/test/java/org/wildfly/test/integration/elytron/sasl/mgmt/ScramPlusMgmtSaslTestCase.java]: @kwart Can these pass now? Elytron was upgraded recently.

**Comment by bstansberry** [comment]
retest this please

**Comment by kwart** [comment]
retest this please

**Pull request merged**